docs/reference/modules/http.asciidoc

```diff
@@ -110,9 +110,16 @@ Which methods to allow. Defaults to `OPTIONS, HEAD, GET, POST, PUT, DELETE`.
 // tag::http-cors-allow-headers-tag[]
 `http.cors.allow-headers` {ess-icon}::
 (<<static-cluster-setting,Static>>)
-Which headers to allow. Defaults to `X-Requested-With, Content-Type, Content-Length`.
+Which headers to allow. Defaults to `X-Requested-With, Content-Type, Content-Length, Authorization, Accept, User-Agent, X-Elastic-Client-Meta`.
```

Comment:
I find it strange that we default `access-control-allow-headers` to include `Authorization` but `access-control-allow-credentials` to `false`?

Is that intentional?

Reply:
Yes, it is intentional:
* [`access-control-allow-headers`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers), in a response to a preflight request, defines what headers (outside of the safelist) are allowed in CORS requests. Since clients set the `Authorization` header in their requests, it has to be allowed by the server in preflight responses.
* Defaulting [`access-control-allow-credentials`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials) to `false` forbids the use of browser-level credentials by clients, and thus requires clients to be provided with credential information that is then set in the `Authorization` header.

CORS is sometimes confusing 😅